test/javascripts/acceptance/search-test.js.es6

```diff
@@ -73,3 +73,53 @@ QUnit.test("Search with context", assert => {
     assert.ok(!$('.search-context input[type=checkbox]').is(":checked"));
   });
 });
+
+QUnit.test("in:likes, in:private, and in:seen filters are hidden to anonymous users", assert => {
+  visit("/search?expanded=true");
+
+  andThen(() => {
+    assert.notOk(exists('.search-advanced-options .in-likes'));
+    assert.notOk(exists('.search-advanced-options .in-private'));
+    assert.notOk(exists('.search-advanced-options .in-seen'));
+  });
+});
+
+QUnit.test("in:likes, in:private, and in:seen filters are available to logged in users", assert => {
+  logIn();
+  Discourse.reset();
+  visit("/search?expanded=true");
+
+  andThen(() => {
+    assert.ok(exists('.search-advanced-options .in-likes'));
+    assert.ok(exists('.search-advanced-options .in-private'));
+    assert.ok(exists('.search-advanced-options .in-seen'));
+  });
+});
+
+QUnit.test("`I've not read`, `I posted in`, `I'm watching`, `I'm tracking`, \
+            `I've bookmarked` filters are hidden to anonymous users from the dropdown", assert => {
+  visit("/search?expanded=true");
+
+  andThen(() => {
+    assert.notOk(exists('select#in option[value=unseen]'));
+    assert.notOk(exists('select#in option[value=posted]'));
+    assert.notOk(exists('select#in option[value=watching]'));
+    assert.notOk(exists('select#in option[value=tracking]'));
+    assert.notOk(exists('select#in option[value=bookmarks]'));
+  });
+});
+
+QUnit.test("`I've not read`, `I posted in`, `I'm watching`, `I'm tracking`, \
+            `I've bookmarked` filters are available to logged in users in the dropdown", assert => {
+  logIn();
+  Discourse.reset();
+  visit("/search?expanded=true");
+
+  andThen(() => {
+    assert.ok(exists('select#in option[value=unseen]'));
+    assert.ok(exists('select#in option[value=posted]'));
+    assert.ok(exists('select#in option[value=watching]'));
+    assert.ok(exists('select#in option[value=tracking]'));
+    assert.ok(exists('select#in option[value=bookmarks]'));
+  });
+});
```

Comment:
The tests may seem a bit redundant. However, IMO, `exists` and `notExists` are both needed to make sure the elements are only rendered differently due to logging in. Open to suggestions on ways to improve testing tho.